pkg/versioning/gomodfile.go

```diff
@@ -91,16 +92,23 @@ func (m *GoMod) GetCoordinates() (Coordinates, error) {
 	if parsed.Module == nil {
 		return result, errors.Wrap(err, "failed to parse go.mod file")
 	}
+
+	// validate module path as defined by golang
+	if err = module.CheckPath(parsed.Module.Mod.Path); err != nil {
+		return result, errors.Wrap(err, "failed to parse go.mod file")
```

Comment:
Can we add a little more detail to this error message? Hinting user it expects a Path?